src/sw.ts

```diff
@@ -199,8 +200,23 @@ async function requestRouting (event: FetchEvent, url: URL): Promise<boolean> {
   } else if (isDeregisterRequest(event.request.url)) {
     event.waitUntil(deregister(event))
     return false
-  } else if (isConfigPageRequest(url) || isSwAssetRequest(event)) {
-    log.trace('config page or sw-asset request, ignoring ', event.request.url)
+  } else if (isConfigPageRequest(url)) {
+    log.trace('config page request, ignoring ', event.request.url)
+    return false
+  } else if (isSwAssetRequest(event)) {
+    log.trace('sw-asset request, returning cached response ', event.request.url)
+    /**
+     * Return the asset from the cache if it exists, otherwise fetch it.
+     */
+    event.respondWith(caches.open(CURRENT_CACHES.swAssets).then(async (cache) => {
+      const cachedResponse = await cache.match(event.request)
+      if (cachedResponse != null) {
+        return cachedResponse
+      }
+      const response = await fetch(event.request)
+      await cache.put(event.request, response.clone())
```

Comment:
One thing to consider is that these won't be evicted and may gradually exhaust the storage quota with every release.

Reply:
We could clear swAssets on "install" just to be safe

Reply:
just confirmation for posterity, install is only fired once per "service worker instance." See https://web.dev/articles/service-worker-lifecycle#install_2

Reply:
fixed by [`a6c3305` (#234)](https://github.com/ipfs-shipyard/service-worker-gateway/pull/234/commits/a6c330534c0c03525ccf1f009638182c1ceaf7b5)